### ISSUE TITLE
Interfacing Arduino to 3G USB modem ( Huawei E3131 ) 

### ISSUE THREAD

**Issue by imedaouidene** [felis/USB_Host_Shield_2.0#301]
I need to interface 3g modem USB dongle ( Huawei E3131)   with my Arduino UNO , To transfer data with high bit rate .
any help ?

**Comment by Blaud** [comment]
For me the best way to get internet on arduino was using nodemcu + [https://www.aliexpress.com/item/Tolkien-LW3G-UIFI-WiFi-Modem-Mini-Portable-Wi-fi-Mobile-Device-3G-Wireless-Router-with-TF/32811968977.html](uri) - usb 3g wifi dongle

**Comment by Blaud** [comment]
@pawisoon It wasn't straight usb connection. I've used a dongle with integrated wifi module, then just power it up with 5v and connect with nodemcu(esp8266) via wifi. It is really simplest and cheapest way to get 3g on your arduino. (this dongle cost about 12$)

**Comment by MohammedZ666** [comment]
@Blaud can u explain how to power up the USB Wifi modem, does it auto-connect to the internet?

**Comment by Blaud** [comment]
@MohammedZ666 yes, it auto connects on power on. I was able to power it by connecting parallel to nodemcu 5v power supply. 
_modem like [this ](https://aliexpress.ru/item/4000595683138.html)_

**Comment by MohammedZ666** [comment]
@Blaud thanks for replying!!
